packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -526,85 +503,399 @@ export const Tabs = (props: TabsProps) => {
-            'iui-animated': type !== 'default' && isClient,
-            'iui-not-animated': type !== 'default' && !isClient,
-            'iui-large': hasSublabel,
-          },
-          tabsClassName,
-        )}
-        data-iui-overflow={overflowOptions?.useOverflow}
-        data-iui-scroll-placement={scrollingPlacement}
-        role='tablist'
-        ref={refs}
-        onKeyDown={onKeyDown}
+        className={cx('iui-tabs-actions-wrapper', className)}
+        ref={ref}
         {...rest}
       >
-        {labels.map((label, index) => createTab(label, index))}
+        {children}
       </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsActions.displayName = 'Tabs.Actions';
 
-      {actions && (
-        <Box className='iui-tabs-actions-wrapper'>
-          <Box className='iui-tabs-actions'>{actions}</Box>
-        </Box>
-      )}
+// ----------------------------------------------------------------------------
+// Tabs.Action component
 
-      {children && (
-        <Box
-          className={cx('iui-tabs-content', contentClassName)}
-          role='tabpanel'
-        >
-          {children}
-        </Box>
-      )}
-    </Box>
+const TabsAction = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+
+  const { type } = useSafeContext(TabsContext);
+
+  if (type !== 'pill') {
+    return (
+      <Box className={cx('iui-tabs-actions', className)} ref={ref} {...rest}>
+        {children}
+      </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsAction.displayName = 'Tabs.Action';
+
+// ----------------------------------------------------------------------------
+// Tabs.Panels component
+
+const TabsPanels = ({
+  children,
+}: {
+  children: React.ReactNode[] | React.ReactNode;
+}) => {
+  const { currentActiveIndex } = useSafeContext(TabsContext);
+
+  const items = Array.isArray(children) ? children : [children];
+
+  return (
+    <>
+      {items.map((item, index) => {
+        return (
+          <TabsContext.Provider
+            value={{ index, currentActiveIndex }}
+            key={index}
+          >
+            {item}
+          </TabsContext.Provider>
+        );
+      })}
+    </>
   );
 };
+TabsPanels.displayName = 'Tabs.Panels';
+
+// ----------------------------------------------------------------------------
+// Tabs.Panel component
+
+const TabsPanel = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+
+  const { index, currentActiveIndex } = useSafeContext(TabsContext);
+
+  if (index === currentActiveIndex) {
+    return (
+      <Box
+        className={cx('iui-tabs-content', className)}
+        role='tabpanel'
+        ref={ref}
+        {...rest}
+      >
+        {children}
+      </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsPanel.displayName = 'Tabs.Panel';
```

Comment:
~~This is for content within the Tab right? I think Tabs.Content would be more a informative name~~

Now that I think about it more, I was expecting to be able to add tab's content under `TabInfo`, along with label, and not in a separate subcomponent. But the original implementation also had them separate :thinking: Regardless I still think `Tabs.Content` would be a clearer name here

Reply:
I like this idea, but does this also mean the `Tabs.Panels` name has to change as well? Perhaps to `Tabs.Contents`?

There's also the idea of keeping `Tabs.Panel` and replacing the wrapper `Tabs.Panels` name with `Tabs.Content`. What are your thoughts @LostABike @gretanausedaite

Reply:
why do we want a generic, undescriptive name like "content"?

"tabpanel" is the canonical, widely-understood term matching the aria role. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tabpanel_role

Reply:
I was just thinking back to Tile's `ContentArea` coupled with my own expectation when using this Tab's component, but wasn't aware of the canonical naming 😅 But TDIL. This is fine then

Reply:
I agree with Mayank here that the `Panel` name is more descriptive and more associated with tabs.

`TabInfo` and `iui-tab-info` have been removed